### ISSUE TITLE
How to allow scrolling on modal if window is too small

### ISSUE THREAD

**Issue by johnwiseheart** [reactjs/react-modal#283]
### Summary:

Hey, I'm having trouble with some styling of this library. I know its not a bug per-se, but figured it might be good to explore if anyone else has had or solved this issue. By default, the styling of the modal is not responsive, in that if you make the window height less than the modal height, you're not able to scroll the modal. This can be fixed by using a fixed height modal, but using a fixed height modal is restrictive on larger screens. 

After some testing, I feel like there is a good solution to this using flexbox, but I haven't managed to find it yet.

TL;DR I'm trying to have a modal fit to its contents, but scroll if the window is too small.

### Expected Behaviour
I'm trying to get the modal to function similarly to this codepen: https://codepen.io/anon/pen/Rodxjx

**Comment by diasbruno** [comment]
@johnwiseheart there is PR #281 which I think It will allow the content of overlay to scroll (modal is inside the overlay).

**Comment by igghera** [comment]
I solved the problem by giving the modal `max-height: 100vh` (that is the height of the viewport) and an `overflow-y: auto`. This way, if the height of the modal is more than the height of the viewport (for example when you bring up the keyboard to fill a text input) the modal adjusts its height to the height of the viewport, allowing you to scroll.

**Comment by corysimmons** [comment]
This fixes it:

```js
ReactModal.defaultStyles = {} // Flushes all of react-modal's styles
```

```css
.ReactModal__Html--open,
.ReactModal__Body--open {
  overflow: hidden; /* prevents background page from scrolling when the modal is open */
}

.ReactModal__Overlay {
  position: fixed;
  z-index: 999999;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  background: rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
}

.ReactModal__Content {
  background: white;
  width: 50rem;
  max-width: calc(100vw - 2rem);
  max-height: calc(100vh - 2rem);
  box-shadow: 0 0 30px 0 rgba(0, 0, 0, 0.25);
  overflow-y: auto;
  position: relative;
}

.modal-close-btn {
  cursor: pointer;
  top: 1.5rem;
  right: 1.5rem;
  position: absolute;
  width: 3rem;
  height: 3rem;
}
```

tbh the maintainers could/should do a major bump and assign `.ReactModal__Overlay`'s styles to `.ReactModalPortal`, then assign `.ReactModal__Content`'s styles to `.ReactModal__Overlay`.

Then they could get rid of the `.ReactModal__Overlay` `<div>` all together while fixing this gross scrolling bug.

**Comment by diasbruno** [comment]
Hey @corysimmons, thanks for having a look at this. 
Can you try to make a PR to show how this would work?

**Comment by corysimmons** [comment]
@diasbruno Yup sure can. Busy the next few days but might have some time over the weekend to prepare something. It's really not much more than what I posted above, but happy to PR. Adding this to my GH Todo. If I forget by this weekend, someone ping me.

**Comment by corysimmons** [comment]
Sorry, still working on work-work stuff... 😩 

Here's what I've got:

<img src="https://user-images.githubusercontent.com/224636/59148839-9ad4ef00-89db-11e9-9169-62609da3f3ce.png" width="200px" />

<img src="https://user-images.githubusercontent.com/224636/59148841-a6281a80-89db-11e9-9980-b6e4c0ca3f58.png" width="200px" />

<img src="https://user-images.githubusercontent.com/224636/59148843-b2ac7300-89db-11e9-8786-dff43287bf22.png" width="200px" />

Code I'm using: https://gist.github.com/corysimmons/c1adcc4af71e7d9eb9ced0437bd208d6

My current approach isn't getting rid of the superfluous wrapping div, but I realized you could potentially have multiple modals (portals) on the screen at the same time so maybe having more = more options for styles for different ones? 🤷‍♂️

**Comment by jackymayo** [comment]
On a side note, onScroll event doesn't seem to be working either.
Is this intended?

**Comment by justinledelson** [comment]
Did anyone ever figure this out?

**Comment by corysimmons** [comment]
@justinledelson It's insane you commented on this. I literally just implemented this and was looking at this issue for my Gist after years. Anyway, my Gist still works great. Just copy/paste it and go from there.

**Comment by diasbruno** [comment]
> @corysimmons's implementation also works, but I'd recommend this way just to separate the modal from the scroll component.

# How to make a scrollable container

`react-modal` is responsible to provide a frame where you can add your content.

To add a scroll, you can add these elements and styles.

```jsx
<Modal ...>
   <div className="scroll-component">
      <div className="scroll-content">
          <Content ... />
      </div>
   </div>
</Modal>
```

The `.scroll-component` must have the save frame size as the `.ReactModal__Content`.

```css
.scroll-component {
   width: 100%;
   height: 100%;
   overflow-y: auto;
}

.scroll-content { /* doesn't need anything... */ }
```

**Comment by diasbruno** [comment]
I'm closing this issue, since it's not react-modal responsibility. Please, feel free to reopen if none of the suggestions works or any questions.

**Comment by pforprince** [comment]
Those who are still wants to scroll the content, add the below piece of code in your CSS file. It worked for me.

```
.ReactModal__Content--after-open {
  overflow-y: scroll !important;
  max-height: 90%;
}
```

**Comment by pforprince** [comment]
> Did anyone ever figure this out?

Those who are still wants to scroll the content, add the below piece of code in your CSS file. It worked for me.

```
.ReactModal__Content--after-open {
  overflow-y: scroll !important;
  max-height: 90%;
}
```

> > @corysimmons's implementation also works, but I'd recommend this way just to separate the modal from the scroll component.
> 
> # How to make a scrollable container
> `react-modal` is responsible to provide a frame where you can add your content.
> 
> To add a scroll, you can add these elements and styles.
> 
> ```js
> <Modal ...>
>    <div className="scroll-component">
>       <div className="scroll-content">
>           <Content ... />
>       </div>
>    </div>
> </Modal>
> ```
> 
> The `.scroll-component` must have the save frame size as the `.ReactModal__Content`.
> 
> ```css
> .scroll-component {
>    width: 100%;
>    height: 100%;
>    overflow-y: auto;
> }
> 
> .scroll-content { /* doesn't need anything... */ }
> ```

Those who are still wants to scroll the content, add the below piece of code in your CSS file. It worked for me.

```
.ReactModal__Content--after-open {
  overflow-y: scroll !important;
  max-height: 90%;
}
```

**Comment by euhansarkar** [comment]
thank you @pforprince .  It's working well. I am using tailwindcss: 
```
<Modal className="absolute border p-3 rounded-lg top-1/4 left-1/2 h-auto w-10/12 lg:w-1/3 bg-white overflow-y-scroll max-h-[90%]">
// code here 
</Modal>
```
